[FEATURE] Additional Text Input Field (#5)

griebie opened the pull request:
Hi Jesper, this is my first pull request on github and yesterday I didn't realize how to manage this. But now I hope you can see my feature and commit. Regards from Hamburg :)

Resolves: #5

sf-tgr committed:
[FEATURE] Additional Text Input Field (#5)

sf-tgr committed:
[FEATURE] Additional Text Input Field (#5)
~ removed debug output in template

bmoex commented on Configuration/TCA/tx_koningbootstrapcarousel_domain_model_slide.php:
```diff
@@ -106,6 +106,32 @@
                 'size' => 30,
             )
         ),
+        'slider_text' => array(
```
Please use 'description' as new rte input

bmoex replied:
I would recommend to add to search fields too (in ```'ctrl'```):
 
```'searchFields' => 'uid, title, description',```

bmoex commented on Configuration/TCA/tx_koningbootstrapcarousel_domain_model_slide.php:
```diff
@@ -106,6 +106,32 @@
             )
         ),
+        'slider_text' => array(
+            'exclude' => 0,
```
'exclude' => true, so permission can be configured per backend user

bmoex commented on Resources/Private/Templates/Carousel/Show.html:
```diff
@@ -24,6 +24,7 @@
                         <h3>
                             <f:format.nl2br>{slide.title}</f:format.nl2br>
                         </h3>
+                        <p><f:format.html>{slide.sliderText}</f:format.html></p>
```
- Add if condition
- ```f:format.html()``` generates own ```<p class="bodytext">``` wrapper

This will result into:
```html
<f:if condition="{slide.description}">
    {slide.description -> f:format.html()}
</f:if>
```

bmoex commented on Classes/Domain/Model/Slide.php:
```diff
@@ -40,6 +40,19 @@ class Slide extends \TYPO3\CMS\Extbase\DomainObject\AbstractEntity
     /**
+     * @var string
+     */
+    protected $sliderText;
```
```protected $description;``` is more generic

bmoex commented on ext_emconf.php:
```diff
@@ -3,7 +3,7 @@
     'description' => 'Carousel with Bootstrap 3.* compliant HTML',
     'category' => 'plugin',
-    'version' => '1.0.1',
+    'version' => '1.1.0',
```
Please dont update version pointer, we will handle releases :-)

bmoex reviewed (changes requested):
Hi @griebie,

Thanks for your pull-request!
Please update for more generic use of new text input field.

Kind regards,

Benjamin Serfhos

sf-tgr committed:
[FEATURE] Additional Text Input Field (#5)
~ renamed slider_text to description in all files
~ inserted if-condition in fluid template
~ reset version number

griebie commented:
Hi bmoex,
thanks for reviewing, learned by that! I changed the naming, the template and reseted the version number with my last commit, hope it fits now, regards,
Thorsten

griebie commented:
Hi @bmoex ,
can you help me out giving a status of this issue? It seems not to be merged into master yet - can I do something for this? 
Thx 4 help - and regards,
Thorsten

bmoex commented:
@jp1987 will merge this and release it on TER :-)
